Add rpms for s390x

jschintag opened the pull request:
Add sandbox and ldd_s390x definitions to rpm/BUILD.bazel. Declare rpmtrees with symlinks and capabilities in rpm/BUILD.bazel. Add centos stream9 repos for s390x.
Run make rpm-deps for s390x.

<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

### What this PR does
Adds the s390x rpm declarations to bazeldnf files, to allow for caching. Does not affect any code or builds, as this is not used by anything in kubevirt yet.
Split off from https://github.com/kubevirt/kubevirt/pull/10490

<!-- (optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: -->
Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:
qemu and libvirt versions are higher than x86 and arm64, as the old ones are no longer available on the centos mirror. They will be bumped by https://github.com/kubevirt/kubevirt/pull/11256

The following alternatives were considered:
Waiting for https://github.com/kubevirt/kubevirt/pull/11256 to merge first. However it is currently blocked.

Links to places where the discussion took place: <!-- optional: slack, other GH issue, mailinglist, ... -->

### Special notes for your reviewer

<!-- optional -->

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [x] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [x] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [x] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [x] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [x] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
NONE
```

jschintag commented:
/cc @andreabolognani 

@andreabolognani I wanted to wait for https://github.com/kubevirt/kubevirt/pull/11256 before doing this, but since it is blocked i decided to go ahead.
I did not change the rpm versions in the script, but instead used enviroment variables to override it. Is that sufficient or should i add special, s390x only, temporary variables for this? My reason being that this way, when a fixed libvirt version is released you can just re-run `make rpm-deps` on your PR and we do not need to revert anything.

andreabolognani commented:
> I wanted to wait for #11256 before doing this, but since it is blocked i decided to go ahead.
> I did not change the rpm versions in the script, but instead used enviroment variables to override it. Is that sufficient or should i add special, s390x only, temporary variables for this? My reason being that this way, when a fixed libvirt version is released you can just re-run `make rpm-deps` on your PR and we do not need to revert anything.

@jschintag can you please explain the rationale for splitting this off the main s390x PR in the first place? Rushing to add s390x RPMs when nothing uses them yet doesn't make a whole lot of sense to me.

Now answering your question: I think we should have the variables, as explicit documentation of the fact that s390x is being treated in a special way. Undoing that as part of my PR is going to be trivial anyway, so don't even worry about that part :)

jschintag commented:
> @jschintag can you please explain the rationale for splitting this off the main s390x PR in the first place? Rushing to add s390x RPMs when nothing uses them yet doesn't make a whole lot of sense to me.

Sure, the main point is to have them start being picked up by the caching job, so that i do not need to keep updating them. Additionally, it will make the main PR easier to review, as there is less that needs to be looked at.

> Now answering your question: I think we should have the variables, as explicit documentation of the fact that s390x is being treated in a special way. Undoing that as part of my PR is going to be trivial anyway, so don't even worry about that part :)

Ok, than i will change it that way. I can also undo it myself after your PR merges.

andreabolognani commented:
> Sure, the main point is to have them start being picked up by the caching job, so that i do not need to keep updating them.

I'm not convinced this is a benefit. You could argue it's the exact opposite: we'd start mirroring to the Google Cloud storage RPMs that are going to become stale before the code that actually needs them is even merged. I won't actively stand in the way of this PR, but personally I would just drop it and let the work happening in #11256 and #10490 run its course. I understand this requires a bit more work on your side, but it still feels like the correct approach to me.

jschintag commented:
> > Sure, the main point is to have them start being picked up by the caching job, so that i do not need to keep updating them.
> 
> I'm not convinced this is a benefit. You could argue it's the exact opposite: we'd start mirroring to the Google Cloud storage RPMs that are going to become stale before the code that actually needs them is even merged. I won't actively stand in the way of this PR, but personally I would just drop it and let the work happening in #11256 and #10490 run its course. I understand this requires a bit more work on your side, but it still feels like the correct approach to me.

Well i think this does have advantages, though it does come with the risk of the packages growing stale before s390x lands upstream.
However it also means that when s390x support lands it will already be in sync with x86/aarch64 and not need to have newer package version because the old ones are no longer on the mirror. And it is not the case that the cached rpms would not be used, they would still support our s390x development effort. It would just not be used upstream (yet).
That being said i agree that there is not really a clear cut "best" approach to the topic.

andreabolognani commented:
> However it also means that when s390x support lands it will already be in sync with x86/aarch64 and not need to have newer package version because the old ones are no longer on the mirror.

This would be a fairly convincing argument if this landed after #11256. I would have no objections at that point.

jschintag commented:
@andreabolognani I have removed the special casing from this PR and updated the rpms.
What do you think, could we merge this so that no matter how long https://github.com/kubevirt/kubevirt/pull/10490 is taking for review, we will already have s390x rpms cached and in sync with the other arches?

andreabolognani commented on hack/rpm-deps.sh:
```diff
@@ -63,6 +63,19 @@ testimage_main="
+  targetcli
+  util-linux
+  which
+"
```
This is now the same as `testimage_main` so there's no point in defining it. Make sure you update the caller below as well.

jschintag replied:
Good catch. I have removed that bit and rebased the PR.
I also needed to move the `qemu-kvm-device-usb-redirect` back to arch specific `launcherbase` definitions, but running `make rpm-deps` after that did not change anything except the sandbox hash.

andreabolognani commented:
@jschintag I just noticed that you didn't include the pr-helper image. Any specific reason for that? It exists on aarch64, and the package seems to be built on s390x as well.

Everything else looks good to me.

jschintag commented:
> @jschintag I just noticed that you didn't include the pr-helper image. Any specific reason for that? It exists on aarch64, and the package seems to be built on s390x as well.
> 
> Everything else looks good to me.

Yeah, it was pointed out to me that it was mostly used for windows VMs, so it did not make much sense to me to include it in the initial enablement.
I might do so in the future, but right now i want to get the core functionality working.

andreabolognani commented:
I don't have a strong opinion one way or another but it still feels weird to me to have this difference, considering that Windows on aarch64 is not much of a thing either, especially when it comes to virtualization...

jschintag commented:
I also don't have a strong opinion on it, which is why i choose the easy way of just skipping it for now. It is easy to enable after all should we need it.

andreabolognani commented:
Fair enough.

/lgtm

jschintag commented:
/assign @jean-edouard

jschintag commented:
/retest-required

jean-edouard reviewed (commented):
/approve
